src/lib.rs

```diff
@@ -131,28 +131,38 @@ pub mod futures;
 #[cfg_attr(docsrs, doc(cfg(feature = "stream")))]
 pub mod stream;
 
-/// Types to configure [`flate2`](::flate2) based encoders.
-#[cfg(feature = "flate2")]
-#[cfg_attr(
-    docsrs,
-    doc(cfg(any(feature = "deflate", feature = "zlib", feature = "gzip")))
-)]
-pub mod flate2 {
-    pub use flate2::Compression;
-}
+mod unshared;
+mod util;
 
-/// Types to configure [`brotli`](::brotli) based encoders.
-#[cfg(feature = "brotli")]
-#[cfg_attr(docsrs, doc(cfg(feature = "brotli")))]
-pub mod brotli {
-    pub use brotli::enc::backward_references::BrotliEncoderParams;
+/// Level of compression data should be compressed with.
+#[derive(Clone, Copy, Debug)]
+pub enum Compression {
```

Comment:
I put it in the crate root, is that okay?

Reply:
Crate root is fine, either name seems fine to me, up to you which to use.

Reply:
I named it `Level`? I hope that's okay.